### ISSUE TITLE
ERROR to install leidenalg in Pycharm

### ISSUE THREAD

**Issue by sitiharyanti** [vtraag/leidenalg#101]
Hi, leidenalg team... I need help to install leidenalg in pycharm using windows system. Python version is 3.8.5.
This the error.

Collecting leidenalg
  Using cached leidenalg-0.8.9.tar.gz (3.8 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: igraph<0.10,>=0.9.0 in c:\users\user\pycharmprojects\1) github-kirito39 22\4\pythonproject_belajar\venv\lib\site-packages (from leidenalg) (0.9.9)
Requirement already satisfied: texttable>=1.6.2 in c:\users\user\pycharmprojects\1) github-kirito39 22\4\pythonproject_belajar\venv\lib\site-packages (from igraph<0.10,>=0.9.0->leidenalg)
 (1.6.4)
Building wheels for collected packages: leidenalg
  Building wheel for leidenalg (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [118 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.10
      creating build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\functions.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\Optimiser.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\version.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\VertexPartition.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\__init__.py -> build\lib.win-amd64-3.10\leidenalg
      running build_ext
      running build_c_core
      -- Building for: Visual Studio 17 2022
      -- Version number: 0.9.6
      -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
      -- The C compiler identification is MSVC 19.31.31105.0
      -- The CXX compiler identification is MSVC 19.31.31105.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64/cl.exe - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Looking for pthread.h
      -- Looking for pthread.h - not found
      -- Found Threads: TRUE
      -- Could NOT find FLEX (missing: FLEX_EXECUTABLE)
      -- Could NOT find BISON (missing: BISON_EXECUTABLE)
      -- Looking for expm1
      -- Looking for expm1 - found
      -- Looking for fmin
      -- Looking for fmin - found
      -- Looking for finite
      -- Looking for finite - not found
      -- Looking for isfinite
      -- Looking for isfinite - found
      -- Looking for log2
      -- Looking for log2 - found
      -- Looking for log1p
      -- Looking for log1p - found
      -- Looking for rint
      -- Looking for rint - found
      -- Looking for rintf
      -- Looking for rintf - found
      -- Looking for round
      -- Looking for round - found
      -- Looking for stpcpy
      -- Looking for stpcpy - not found
      -- Looking for strcasecmp
      -- Looking for strcasecmp - not found
      -- Looking for strdup
      -- Looking for strdup - found
      -- Looking for _stricmp
      -- Looking for _stricmp - found
      -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
      -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Failed
      -- Performing Test COMPILER_HAS_DEPRECATED
      -- Performing Test COMPILER_HAS_DEPRECATED - Success
      CMake Error at CMakeLists.txt:134 (add_subdirectory):
        add_subdirectory given source "interfaces" which is not an existing
        directory.
     
     
      -- Performing Test HAVE_ENABLE_NEW_DTAGS
      -- Performing Test HAVE_ENABLE_NEW_DTAGS - Failed
      -- Found Python3: C:/Users/user/PycharmProjects/1) Github-Kirito39 22/4/pythonProject_BELAJAR/venv/Scripts/python.exe (found version "3.10.3") found components: Interpreter
      --
      -- -----[ Build configuration ]----
      -- Version:                   0.9.6
      -- CMake build type:        default
      -- Library type:             static
      --
      -- ----------[ Features ]----------
      -- GLPK for optimization:       yes
      -- Reading GraphML files:       yes
      -- Thread-local storage:         no
      -- Link-time optimization:       no
      --
      -- --------[ Dependencies ]--------
      -- ARPACK:                 vendored
      -- BISON:                        no
      -- BLAS:                   vendored
      -- CXSparse:               vendored
      -- FLEX:                         no
      -- GLPK:                   vendored
      -- GMP:                    vendored
      -- LAPACK:                 vendored
      -- LibXml2:                     yes
      -- OpenMP:                      yes
      -- PLFIT:                  vendored
      --
      -- -----------[ Testing ]----------
      -- Diff tool:                    fc
      -- Sanitizers:                 none
      -- Code coverage:                no
      -- Verify 'finally' stack:       no
      --
      -- --------[ Documentation ]-------
      -- HTML:                         no
      -- PDF:                          no
      --
      -- igraph configured successfully.
      --
      -- Configuring incomplete, errors occurred!
      See also "C:/Users/user/AppData/Local/Temp/pip-install-42_ac89i/leidenalg_21c6663a28b64edaa7c52c05084a16db/vendor/build/igraph/CMakeFiles/CMakeOutput.log".
      See also "C:/Users/user/AppData/Local/Temp/pip-install-42_ac89i/leidenalg_21c6663a28b64edaa7c52c05084a16db/vendor/build/igraph/CMakeFiles/CMakeError.log".
      error: [Errno 2] No such file or directory: 'C:\\Users\\user\\AppData\\Local\\Temp\\pip-install-42_ac89i\\leidenalg_21c6663a28b64edaa7c52c05084a16db\\vendor\\install\\igraph\\build.
cfg'
      We are going to build the C core of igraph.
        Source folder: vendor\source\igraph
        Build folder: vendor\build\igraph
        Install folder: vendor\install\igraph
     
      Configuring build...
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for leidenalg
  Running setup.py clean for leidenalg
Failed to build leidenalg
Installing collected packages: leidenalg
  Running setup.py install for leidenalg ... error
  error: subprocess-exited-with-error

  × Running setup.py install for leidenalg did not run successfully.
  │ exit code: 1
  ╰─> [69 lines of output]
      running install
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-3.10
      creating build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\functions.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\Optimiser.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\version.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\VertexPartition.py -> build\lib.win-amd64-3.10\leidenalg
      copying src\leidenalg\__init__.py -> build\lib.win-amd64-3.10\leidenalg
      running build_ext
      running build_c_core
      -- Version number: 0.9.6
      -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
      -- Could NOT find FLEX (missing: FLEX_EXECUTABLE)
      -- Could NOT find BISON (missing: BISON_EXECUTABLE)
      CMake Error at CMakeLists.txt:134 (add_subdirectory):
        add_subdirectory given source "interfaces" which is not an existing
        directory.
     
     
      --
      -- -----[ Build configuration ]----
      -- Version:                   0.9.6
      -- CMake build type:        default
      -- Library type:             static
      --
      -- ----------[ Features ]----------
      -- GLPK for optimization:       yes
      -- Reading GraphML files:       yes
      -- Thread-local storage:         no
      -- Link-time optimization:       no
      --
      -- --------[ Dependencies ]--------
      -- ARPACK:                 vendored
      -- BISON:                        no
      -- BLAS:                   vendored
      -- CXSparse:               vendored
      -- FLEX:                         no
      -- GLPK:                   vendored
      -- GMP:                    vendored
      -- LAPACK:                 vendored
      -- LibXml2:                     yes
      -- OpenMP:                      yes
      -- PLFIT:                  vendored
      --
      -- -----------[ Testing ]----------
      -- Diff tool:                    fc
      -- Sanitizers:                 none
      -- Code coverage:                no
      -- Verify 'finally' stack:       no
      --
      -- --------[ Documentation ]-------
      -- HTML:                         no
      -- PDF:                          no
      --
      -- igraph configured successfully.
      --
      -- Configuring incomplete, errors occurred!
      See also "C:/Users/user/AppData/Local/Temp/pip-install-42_ac89i/leidenalg_21c6663a28b64edaa7c52c05084a16db/vendor/build/igraph/CMakeFiles/CMakeOutput.log".
      See also "C:/Users/user/AppData/Local/Temp/pip-install-42_ac89i/leidenalg_21c6663a28b64edaa7c52c05084a16db/vendor/build/igraph/CMakeFiles/CMakeError.log".
      error: [Errno 2] No such file or directory: 'C:\\Users\\user\\AppData\\Local\\Temp\\pip-install-42_ac89i\\leidenalg_21c6663a28b64edaa7c52c05084a16db\\vendor\\install\\igraph\\build.
cfg'
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> leidenalg

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

**Comment by vtraag** [comment]
For some reason you are building from source. What is the reason you can't install a binary wheel from PyPI?

At any rate, in order to build from source, you additionally still need the `bison` and `flex` dependencies. You can install them in windows using Chocolatey: `choco install winflexbison3`.

If that doesn't solve your problem, let me know.

**Comment by mlaponsky** [comment]
Also encountering this issue, and it happens when installing from PyPI (`pip install leidenalg`). Seems that whatever gets pulled from pip, it still tries to build the wheel and fails at that step. I also installed `bison` and `flex` ahead of time, no luck. And this is on a Mac, not a Windows machine, so it doesn't seem to be specific to OS or to PyCharm.

**Comment by licycle** [comment]
same issue.installed bison and flex does not work.
I am trying to install leidenalg on a linux arm server.

**Comment by mlaponsky** [comment]
Can also confirm that the same error presents when trying to install in the `python:3.8` Docker image.

**Comment by vtraag** [comment]
Could you both @mlaponsky and @licycle please provide some more detail and include the output when you try to install it? 

There are currently no ARM binaries, so that is perhaps why you don't get binaries from PyPI. Providing binaries from PyPI for that case is a bit more involved, as it requires some cross-compilation, but I'll try to look into it. Perhaps you can install `leidenalg` using `conda`? There are binaries for ARM supplied on the `conda-forge` channel.

**Comment by mlaponsky** [comment]
I don't think it's ARM-specific, but may be an issue with what's up on PyPI. I just ran the `pip install` on an Intel Mac with the same error. **However**, `conda install -c conda-forge leidenalg` worked on my M1 Mac and Intel Mac. Here's my full error stack for the pip install:
```
pip install leidenalg
Collecting leidenalg
  Using cached leidenalg-0.8.9.tar.gz (3.8 MB)
Requirement already satisfied: igraph<0.10,>=0.9.0 in ./.pyenv/versions/3.10.0/envs/lsh/lib/python3.10/site-packages (from leidenalg) (0.9.10)
Requirement already satisfied: texttable>=1.6.2 in ./.pyenv/versions/3.10.0/envs/lsh/lib/python3.10/site-packages (from igraph<0.10,>=0.9.0->leidenalg) (1.6.4)
Building wheels for collected packages: leidenalg
  Building wheel for leidenalg (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"'; __file__='"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-wheel-r_gt31tr
       cwd: /private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/
  Complete output (129 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-12.0-x86_64-3.10
  creating build/lib.macosx-12.0-x86_64-3.10/leidenalg
  copying src/leidenalg/functions.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
  copying src/leidenalg/Optimiser.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
  copying src/leidenalg/VertexPartition.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
  copying src/leidenalg/version.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
  copying src/leidenalg/__init__.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
  running build_ext
  running build_c_core
  We are going to build the C core of igraph.
    Source folder: vendor/source/igraph
    Build folder: vendor/build/igraph
    Install folder: vendor/install/igraph

  Configuring build...
  -- Setting build type to 'Release' as none was specified.
  -- Version number: 0.9.6
  -- The C compiler identification is AppleClang 13.1.6.13160021
  -- The CXX compiler identification is AppleClang 13.1.6.13160021
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Performing Test COMPILER_SUPPORTS_NO_VARARGS_FLAG
  -- Performing Test COMPILER_SUPPORTS_NO_VARARGS_FLAG - Success
  -- Performing Test COMPILER_SUPPORTS_NO_UNKNOWN_WARNING_OPTION_FLAG
  -- Performing Test COMPILER_SUPPORTS_NO_UNKNOWN_WARNING_OPTION_FLAG - Success
  -- Looking for pthread.h
  -- Looking for pthread.h - found
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
  -- Found Threads: TRUE
  -- Found FLEX: /usr/bin/flex (found version "2.6.4")
  -- Found BISON: /usr/bin/bison (found version "2.3")
  -- Looking for expm1
  -- Looking for expm1 - found
  -- Looking for fmin
  -- Looking for fmin - found
  -- Looking for finite
  -- Looking for finite - found
  -- Looking for isfinite
  -- Looking for isfinite - found
  -- Looking for log2
  -- Looking for log2 - found
  -- Looking for log1p
  -- Looking for log1p - found
  -- Looking for rint
  -- Looking for rint - found
  -- Looking for rintf
  -- Looking for rintf - found
  -- Looking for round
  -- Looking for round - found
  -- Looking for stpcpy
  -- Looking for stpcpy - found
  -- Looking for strcasecmp
  -- Looking for strcasecmp - found
  -- Looking for strdup
  -- Looking for strdup - found
  -- Looking for _stricmp
  -- Looking for _stricmp - not found
  -- Looking for _LIBCPP_VERSION
  -- Looking for _LIBCPP_VERSION - found
  -- Looking for __GLIBCXX__
  -- Looking for __GLIBCXX__ - not found
  -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
  -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
  -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
  -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
  -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
  -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
  CMake Error at CMakeLists.txt:134 (add_subdirectory):
    add_subdirectory given source "interfaces" which is not an existing
    directory.


  -- Performing Test HAVE_ENABLE_NEW_DTAGS
  -- Performing Test HAVE_ENABLE_NEW_DTAGS - Failed
  -- Found Python3: /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/bin/python3.10 (found version "3.10.0") found components: Interpreter
  --
  -- -----[ Build configuration ]----
  -- Version:                   0.9.6
  -- CMake build type:        Release
  -- Library type:             static
  --
  -- ----------[ Features ]----------
  -- GLPK for optimization:       yes
  -- Reading GraphML files:       yes
  -- Thread-local storage:         no
  -- Link-time optimization:       no
  --
  -- --------[ Dependencies ]--------
  -- ARPACK:                 vendored
  -- BISON:                       yes
  -- BLAS:                   vendored
  -- CXSparse:               vendored
  -- FLEX:                        yes
  -- GLPK:                   vendored
  -- GMP:                    vendored
  -- LAPACK:                 vendored
  -- LibXml2:                     yes
  -- OpenMP:                       no
  -- PLFIT:                  vendored
  --
  -- -----------[ Testing ]----------
  -- Diff tool:                  diff
  -- Sanitizers:                 none
  -- Code coverage:                no
  -- Verify 'finally' stack:       no
  --
  -- --------[ Documentation ]-------
  -- HTML:                         no
  -- PDF:                          no
  --
  -- igraph configured successfully.
  --
  -- Configuring incomplete, errors occurred!
  See also "/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/build/igraph/CMakeFiles/CMakeOutput.log".
  See also "/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/build/igraph/CMakeFiles/CMakeError.log".
  error: [Errno 2] No such file or directory: '/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/install/igraph/build.cfg'
  ----------------------------------------
  ERROR: Failed building wheel for leidenalg
  Running setup.py clean for leidenalg
Failed to build leidenalg
Installing collected packages: leidenalg
    Running setup.py install for leidenalg ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"'; __file__='"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-record-i9eihucg/install-record.txt --single-version-externally-managed --compile --install-headers /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/include/site/python3.10/leidenalg
         cwd: /private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/
    Complete output (66 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-12.0-x86_64-3.10
    creating build/lib.macosx-12.0-x86_64-3.10/leidenalg
    copying src/leidenalg/functions.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
    copying src/leidenalg/Optimiser.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
    copying src/leidenalg/VertexPartition.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
    copying src/leidenalg/version.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
    copying src/leidenalg/__init__.py -> build/lib.macosx-12.0-x86_64-3.10/leidenalg
    running build_ext
    running build_c_core
    We are going to build the C core of igraph.
      Source folder: vendor/source/igraph
      Build folder: vendor/build/igraph
      Install folder: vendor/install/igraph

    Configuring build...
    -- Version number: 0.9.6
    CMake Error at CMakeLists.txt:134 (add_subdirectory):
      add_subdirectory given source "interfaces" which is not an existing
      directory.


    --
    -- -----[ Build configuration ]----
    -- Version:                   0.9.6
    -- CMake build type:        Release
    -- Library type:             static
    --
    -- ----------[ Features ]----------
    -- GLPK for optimization:       yes
    -- Reading GraphML files:       yes
    -- Thread-local storage:         no
    -- Link-time optimization:       no
    --
    -- --------[ Dependencies ]--------
    -- ARPACK:                 vendored
    -- BISON:                       yes
    -- BLAS:                   vendored
    -- CXSparse:               vendored
    -- FLEX:                        yes
    -- GLPK:                   vendored
    -- GMP:                    vendored
    -- LAPACK:                 vendored
    -- LibXml2:                     yes
    -- OpenMP:                       no
    -- PLFIT:                  vendored
    --
    -- -----------[ Testing ]----------
    -- Diff tool:                  diff
    -- Sanitizers:                 none
    -- Code coverage:                no
    -- Verify 'finally' stack:       no
    --
    -- --------[ Documentation ]-------
    -- HTML:                         no
    -- PDF:                          no
    --
    -- igraph configured successfully.
    --
    -- Configuring incomplete, errors occurred!
    See also "/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/build/igraph/CMakeFiles/CMakeOutput.log".
    See also "/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/build/igraph/CMakeFiles/CMakeError.log".
    error: [Errno 2] No such file or directory: '/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/vendor/install/igraph/build.cfg'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"'; __file__='"'"'/private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-install-sv2zqli4/leidenalg_ca54a02d76894dc3aa2e7356d1fa9cb7/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/7p/h5jyx1397y98pvfvh3cn9sz00000gn/T/pip-record-i9eihucg/install-record.txt --single-version-externally-managed --compile --install-headers /Users/maxlaponsky/.pyenv/versions/3.10.0/envs/lsh/include/site/python3.10/leidenalg Check the logs for full command output.```

**Comment by vtraag** [comment]
Thanks for the details. In this case, you are trying to install `leidenalg` for Python 3.10, but when the latest version was released, only binaries up until Python 3.9 were included.

**Comment by vtraag** [comment]
I located the problem, I accidentally forgot to no longer prune the `interfaces` directory, this is now fixed in 4f7c9f22701ce83f4788b8a7196fc472932025e2.

**Comment by vtraag** [comment]
Here's the source distribution that should work again:
[leidenalg-0.8.10.dev4+gcc5f27c.tar.gz](https://github.com/vtraag/leidenalg/files/8479655/leidenalg-0.8.10.dev4%2Bgcc5f27c.tar.gz).

I'm currently upgrading some other things to make sure that binary wheels are at least also available for Python 3.10. I'll also explore whether I can make binary wheels available for ARM.  When that's done, I'll make a new release that should prevent these problems.

**Comment by pcarbo** [comment]
@vtraag We are also experiencing the same issues (on a CentOS system) and we look forward to the fixes. Thanks.

**Comment by paginabianca** [comment]
Had the same issue on Archlinux, 4f7c9f2 fixes it. Thank you :+1:

**Comment by vtraag** [comment]
I just released version 0.8.10, which should resolve problems when installing from source. It also provides binary wheels for Python 3.10 and binary wheels for ARM (macOS) and aarch64 (Linux).
